modules/dxp/apps/sharepoint-soap/sharepoint-soap-repository/build.gradle

```diff
@@ -11,7 +11,6 @@ buildWSDL {
 }
 
 dependencies {
-	compileInclude group: "com.liferay", name: "org.apache.commons.httpclient", version: "3.1.LIFERAY-PATCHED-1"
```

Comment:
Hi @jeffwu0724, you may upgraded the wrong jar.

**com.liferay:org.apache.commons.httpclient**:3.1.LIFERAY-PATCHED-1 matches with third-party\org-apache-commons-httpclient. And the third party should **commons-httpclient:commons-httpclient:3.1**.

**org.apache.httpcomponents:httpclient** is related with our third-party\org-apache-httpcomponents-httpclient. https://issues.liferay.com/browse/LPS-134639 resolved the issue.

Reply:
Hi @yuhai , 

I have some questions about this:

- for the patch **/third-party/org-apache-httpcomponents-httpclient - 4.1.3.LIFERAY-PATCHED-1**, seems like nobody is using it now in master. Should we remove this patch and make a new patch based with the latest version of HttpClient, for example, 4.5.13 in this case?

- When we update usage in sharepoint-soap-repository based on https://issues.liferay.com/browse/LPS-154824, should we create a new patch with version 4.5.13, or should we just use this third party jar directly?

Thanks for your help!

Reply:
Hi @jeffwu0724,

1. third-party only be maintained on master. I noticed 7.1.x and 7.0.x still used this patched version. So we can't remove the patch version. If liferay versions 7.0+ (we may also consider 6.2.x) didn't use the patch, I think we may remove it. 

2. At this case, we needn't to make the patch for 4.5.13 because 4.5.13 doesn't exist security issue.  You may refer to https://mvnrepository.com/artifact/org.apache.httpcomponents/httpclient. Since https://issues.liferay.com/browse/LPS-134639 upgrade it to 4.5.13, we may backport it. After that, I think we may remove the patch version **org-apache-httpcomponents-httpclient - 4.1.3.LIFERAY-PATCHED-1**. By the way, I just left comment to ask if we can use LPS-134639 to do the backport.

Reply:
Hi @yuhai ,

Thanks for your explanation! 
Just to confirm, for this ticket:

1. we do not need to make any change to **org.apache.commons.httpclient - 3.1.LIFERAY-PATCHED-1**, and just keep it for now.
2. We just need to wait for Samuel's response and see if we can backport LPS-134639 to solve this ticket. 
3. After everything is done, then we will have to remove **org-apache-httpcomponents-httpclient - 4.1.3.LIFERAY-PATCHED-1** since it is not needed anymore

Thanks for your reply in advance! :)

Reply:
Hi @jeffwu0724，

1. Yes
2. Yes
3. Yes, please create one new ticket for this after everything is done.

Reply:
Hi @yuhai ,

Got it, thanks a lot! I will wait for Samuel's response in https://issues.liferay.com/browse/LPS-154824, and prepare for the next step. For now I will close this ticket first. 
Really appreciate your help!